File: rei-do-chopp-api/ReiDoChopp.DataTransfer/Users/Requests/UserEditRequest.cs
using System.ComponentModel.DataAnnotations;

namespace ReiDoChopp.DataTransfer.Users.Requests
{
    public class UserEditRequest
    {
        [Required]
        [EmailAddress]
        public string Email { get; set; }

        [Required]
        public string FirstName { get; set; }

        [Required]
        public string LastName { get; set; }

        public int[] RoleIds { get; set; }

    }
}

File: rei-do-chopp-site/src/app/users/models/requests/user-edit-request.ts
export class UserEditRequest {
  Email: string;
  FirstName: string;
  LastName: string;
  RoleIds: number[];

  constructor(params: Partial<UserEditRequest> = {}) {
    this.Email = params.Email ?? '';
    this.FirstName = params.FirstName ?? '';
    this.LastName = params.LastName ?? '';
    this.RoleIds = params.RoleIds || [];
  }
}
